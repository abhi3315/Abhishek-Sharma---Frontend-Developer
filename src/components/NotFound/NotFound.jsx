import { Link } from 'react-router-dom';

function NotFound() {
	return (
		<main className='h-screen w-full flex flex-col justify-center items-center'>
			<h1 className='text-9xl font-extrabold text-gray-800 tracking-widest dark:text-gray-200'>
				404
			</h1>
			<div className='bg-red-400 px-2 text-sm rounded rotate-12 absolute'>Page Not Found</div>
			<Link
				to='/'
				className='text-gray-500 dark:text-gray-400 hover:bg-gray-200 dark:hover:bg-zinc-600 focus:outline-none focus:ring-4 focus:ring-gray-300 dark:focus:ring-zinc-500 rounded-lg text-xl p-2.5 inline-flex items-center'
			>
				Go Home
			</Link>
		</main>
	);
}

export default NotFound;
