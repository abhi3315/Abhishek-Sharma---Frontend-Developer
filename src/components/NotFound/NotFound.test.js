import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import NotFound from './NotFound';

afterEach(cleanup);

describe('NotFound', () => {
	it('should render correctly', () => {
		const { container } = render(<NotFound />, {
			wrapper: BrowserRouter,
		});
		expect(container).toMatchSnapshot();
	});
});
