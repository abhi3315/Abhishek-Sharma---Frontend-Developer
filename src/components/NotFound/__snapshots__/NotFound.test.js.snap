// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render correctly 1`] = `
<div>
  <main
    class="h-screen w-full flex flex-col justify-center items-center"
  >
    <h1
      class="text-9xl font-extrabold text-gray-800 tracking-widest dark:text-gray-200"
    >
      404
    </h1>
    <div
      class="bg-red-400 px-2 text-sm rounded rotate-12 absolute"
    >
      Page Not Found
    </div>
    <a
      class="text-gray-500 dark:text-gray-400 hover:bg-gray-200 dark:hover:bg-zinc-600 focus:outline-none focus:ring-4 focus:ring-gray-300 dark:focus:ring-zinc-500 rounded-lg text-xl p-2.5 inline-flex items-center"
      href="/"
    >
      Go Home
    </a>
  </main>
</div>
`;
